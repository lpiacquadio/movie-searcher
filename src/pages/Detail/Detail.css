.Detail {
    display: flex;
    flex-direction: column;
    gap: 20px;
    padding: 0px 20px;
}

.Detail-Content {
    display: flex;
    flex-direction: column;
    gap: 20px;
}
