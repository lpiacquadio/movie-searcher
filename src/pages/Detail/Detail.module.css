.Detail {
    display: flex;
    flex-direction: column;
    gap: 20px;
}

.Detail-Content {
    display: flex;
    flex-direction: column;
    gap: 20px;
}
