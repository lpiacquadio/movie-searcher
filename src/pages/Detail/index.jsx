import React from 'react'

export function Detail() {
    return <div>Detail Page</div>
}
