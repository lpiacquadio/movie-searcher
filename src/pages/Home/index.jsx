import React from 'react'
import { useShows } from '../../hooks/useShows'
import { Search } from '../../components/Search'
import { Spinner } from '../../components/Spinner'
import { Info } from '../../components/Info'
import { ListOfShows } from '../../components/ListOfShows'
import { DEFAULT_TEXTS } from '../../services/settings'
import './Home.css'

export function Home() {
    const { loading, shows, search, onSearch } = useShows('')
    return (
        <div>
            <header className="Searcher">
                <div className="Title">{DEFAULT_TEXTS.TITLE}</div>
                <Search
                    onSearch={onSearch}
                    placeholder={DEFAULT_TEXTS.SEARCH_PLACEHOLDER}
                    value={search}
                />
            </header>
            <main className="Content">
                {loading ? (
                    <Spinner />
                ) : (
                    <ListOfShows
                        onNotStarted={<Info>{DEFAULT_TEXTS.NOT_STARTED}</Info>}
                        onNotFound={<Info>{DEFAULT_TEXTS.NOT_FOUND}</Info>}
                        shows={shows}
                        search={search}
                    />
                )}
            </main>
        </div>
    )
}
