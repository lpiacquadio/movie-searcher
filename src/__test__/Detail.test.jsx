import React from 'react'
import renderer from 'react-test-renderer'
import { describe, expect, test } from 'vitest'
import { Detail } from '../pages/Detail'

describe('Detail', () => {
    test('Detail component renders correctly', () => {
        const component = renderer.create(<Detail />)
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})
