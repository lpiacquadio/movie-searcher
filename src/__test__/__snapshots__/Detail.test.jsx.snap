// Vitest Snapshot v1

exports[`Detail > Detail component renders correctly 1`] = `
<div>
  Detail Page
</div>
`;
